BaseUnit.cpp

```diff
@@ -19,6 +21,83 @@ std::string BaseUnit::showStats() const
 {
 	return Name + ": HP:" + std::to_string(HP) + " DMG: " + std::to_string(DMG);
 }
+BaseUnit BaseUnit::parseUnit(const std::string& file_name)
+{
+	std::fstream infile(file_name);
+
+	if (!infile.is_open())
+	{
+		throw(file_name);
+	}
+
+	std::string line;
+	std::string nm = "";
+	int hp = -1;
+	int dm = -1;
+
+	//loop to identify unit attributes
+	while (!infile.eof())
+	{
+		std::getline(infile, line);
+
+		size_t pos;
+
+		//find name
+		if (nm == "")
+		{
+			pos = line.find("name");
+
+			if (pos != std::string::npos)
+			{
+				pos = line.find(":");
+				nm = line.substr(pos + 3, line.size());
+				nm.pop_back();
+				nm.pop_back();
+
+				continue;
+			}
+		}
+		//find hp
+		if (hp == -1)
+		{
+			pos = line.find("hp");
+			if (pos != std::string::npos)
```

Comment:
Azért ez így nem túl pontos parsolás. Ez elfogadja a `"nemhp" : 420` sort is. És a vesszőket sem veszi figyelembe.

Reply:
A `"nemhp"` kijavítására van ötletem, ezt a vesszős dolgot viszont nem értem. Itt arra gondolsz, hogy a vessző jelentené a sor végét és odáig olvassam be, vagy valami egészen másra?

Reply:
Arra gondoltam, hogy elfogadja azt is, ha a key-value párok felsorolásában nincs vessző a párok közt. És igen, lehetne használni beolvasásnál is elválasztójelként, csak általános esetben a vessző idézőjelek között is lehet egy szöveges adatban, erre vigyázni kell.
Most ezt a vesszős dolgot hagyjuk, de azért a kulcs string rendesen legyen beolvasva, és a névnél lehessen akármennyi szóköz a `:` után.